server/src/test/java/org/elasticsearch/indices/cluster/ClusterStateChanges.java

```diff
@@ -140,9 +144,29 @@ public ClusterStateChanges(NamedXContentRegistry xContentRegistry, ThreadPool th
         DestructiveOperations destructiveOperations = new DestructiveOperations(settings, clusterSettings);
         Environment environment = TestEnvironment.newEnvironment(settings);
         Transport transport = mock(Transport.class); // it's not used
+        nextMasterTaskToRun = new AtomicReference<>();
+        FakeThreadPoolMasterService masterService = new FakeThreadPoolMasterService("fake-master", nextMasterTaskToRun::set);
```

Comment:
Does this need any kind of assertion that `nextMasterTaskToRun` isn't already set?